src/hooks/useTxDetails/useTxDetails.ts

```diff
@@ -25,12 +25,12 @@ enum Direction {
 
 export interface TxDetails {
   tx: Tx
-  buyTx: TxTransfer | undefined
-  sellTx: TxTransfer | undefined
-  tradeTx: TxTransfer | undefined
-  feeAsset: Asset
-  buyAsset: Asset
-  sellAsset: Asset
+  buyTx?: TxTransfer
+  sellTx?: TxTransfer
+  tradeTx?: TxTransfer
+  feeAsset?: Asset
+  buyAsset?: Asset
+  sellAsset?: Asset
```

Comment:
Just want to make sure, are these either passed with a value, or not at all?
A value not existing in an object (where `Object.prototype.hasOwnProperty()` would return false) and a value existing as `undefined` are two different things

Reply:
Yes indeed - confirming I've intentionally changed to a wider type, e.g. can be `undefined` or omitted entirely to better support generic transactions in the future.